[Docs] Add a single-point starter guide for contributors (#21322)
openvinotoolkit/openvino: CONTRIBUTING.md

     share your expertise with the community. Check GitHub Discussions and 
     Issues to see if you can help someone. 

## Technical guide

> This section lists all the necessary steps required to setup your environment, build OpenVINO locally and run tests for specific components. It's a perfect place to start when you have just picked a Good First Issue and are wondering how to start working on it.

Reviewer: ```suggestion
This section lists all the necessary steps required to set up your environment, build OpenVINO locally, and run tests for specific components. It's a perfect place to start when you have just picked a Good First Issue and are wondering how to start working on it.
```

> Keep in mind that we are here to help - **do not hesitate to ask the development team if something is not clear**. Such questions allow up to keep improving our documentation.

Reviewer: ```suggestion
Keep in mind that we are here to help - **do not hesitate to ask the development team if something is not clear**. Such questions allow us to keep improving our documentation.
```

### 1. Prerequisites 

You can start with the following links:
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [OpenVINO architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/docs/architecture.md)
- [User documentation](https://docs.openvino.ai/)
- [Blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)
- [Pick up a Good First Issue](https://github.com/orgs/openvinotoolkit/projects/3)

### 2. Building the project

> In order to build the project, please follow the [build instructions for your specific OS](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md).

Reviewer: ```suggestion
In order to build the project, follow the [build instructions for your specific OS](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md).
```

> ### 3. Familiarize yourself with component you'll be working with

Reviewer: ```suggestion
### 3. Familiarize yourself with the component you'll be working with
```

> Choose the component your Good First Issue is related to. You can run tests to make sure it is working correctly.

Reviewer: ```suggestion
Choose the component your Good First Issue is related to. You can run tests to make sure it works correctly.
```

##### Plugins
- [Auto plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/auto)
- [CPU plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu)
- [GPU plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu)
- [Hetero plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/hetero)
- [Template plugin](https://github.com/openvinotoolkit/openvino/tree/master/src/plugins/template)

##### Tools
- [Benchmark Tool](https://github.com/openvinotoolkit/openvino/tree/master/tools/benchmark_tool)
- [Model Optimizer](https://github.com/openvinotoolkit/openvino/tree/master/tools/mo)

##### Others
- [Documentation](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING_DOCS.md)

### 3. Start working on your Good First Issue

Use the issue description and locally built OpenVINO to complete the task. Remember that you can always ask users tagged in the "Contact points" section for help!

### 4. Submit a PR with your changes

Follow our [Good Pull Request guidelines](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING_PR.md).

> ### 5. Wait for review

Reviewer: ```suggestion
### 5. Wait for a review
```
